bag/layout/objects.py

```diff
@@ -494,10 +494,10 @@ def blockage_iter(self, layer_id, test_box, spx=0, spy=0):
         yb = base_box.bottom_unit
         xr = base_box.right_unit
         yt = base_box.top_unit
-        nx0 = max(0, -(-(test.left_unit - xr) // inst_spx))
-        nx1 = min(self.nx - 1, (test.right_unit - xl) // inst_spx)
-        ny0 = max(0, -(-(test.bottom_unit - yt) // inst_spy))
-        ny1 = min(self.ny - 1, (test.top_unit - yb) // inst_spy)
+        nx0 = int(max(0, -(-(test.left_unit - xr) // inst_spx)))
+        nx1 = int(min(self.nx - 1, (test.right_unit - xl) // inst_spx))
+        ny0 = int(max(0, -(-(test.bottom_unit - yt) // inst_spy)))
+        ny1 = int(min(self.ny - 1, (test.top_unit - yb) // inst_spy))
```

Comment:
Don't put 'int' here. If self.nx and self.ny are not integers, then bug is somewhere else.